R/swagger.R

```diff
@@ -4,15 +4,34 @@
 swaggerTypeInfo <- list()
 plumberToSwaggerTypeMap <- list()
 defaultSwaggerType <- "string"
+attr(defaultSwaggerType, "default") <- TRUE
+defaultSwaggerIsArray <- FALSE
+attr(defaultSwaggerIsArray, "default") <- TRUE
 
 local({
-  addSwaggerInfo <- function(swaggerType, plumberTypes, regex, converter) {
+  addSwaggerInfo <- function(swaggerType, plumberTypes,
+                             regex = NULL, converter = NULL,
+                             format = NULL,
+                             location = NULL,
+                             realType = NULL,
+                             arraySupport = FALSE) {
     swaggerTypeInfo[[swaggerType]] <<-
       list(
         regex = regex,
-        converter = converter
+        converter = converter,
+        format = format,
+        location = location,
+        arraySupport = arraySupport,
+        realType = realType
       )
 
+    if (arraySupport == TRUE) {
+      swaggerTypeInfo[[swaggerType]] <<- modifyList(
+        swaggerTypeInfo[[swaggerType]],
+        list(regexArray = paste0("(?:(?:", regex, "),?)+"),
+             converterArray = function(x) {converter(stri_split_fixed(x, ",")[[1]])})
```

Comment:
I don't know if this will handle `c("A,a", "B")`. 

I don't know of a clean solution. Will take some digging, maybe in a second PR later.

Reply:
In the path you mean?

Reply:
`/test/A%2Ca,B/testy`

Something like that

Reply:
If the string value has a comma, but is an item array... can we determine that?

Reply:
Which item are arrays is defined when the router is created. Array converter will only be used on predefined array path parameters.

If you want to use a coma in a value that is part of an array path parameter, you would have to URLEncode the coma first (%2C).

Reply:
> Additionally, the allowReserved keyword specifies whether the reserved characters :/?#[]@!$&'()*+,;= in parameter values are allowed to be sent as they are, or should be percent-encoded. By default, allowReserved is false, and reserved characters are percent-encoded. For example, / is encoded as %2F (or %2f), so that the parameter value quotes/h2g2.txt will be sent as quotes%2Fh2g2.txt
https://swagger.io/docs/specification/serialization/

It is the same if you want a `/` in your string, you have to encode it.

Reply:
Should we be url decoding the values as well, then?

Reply:
Ah. For now, we will not `allowReserved`.

Thank you!

Reply:
Probably? It would be in `extractPathParams`

Reply:
For now, I think we should not. 

Until we add an option to the route to declare that params should be decoded.

Reply:
What tickles me abit is that query parameters are decoded but not class converted, path parameters are not decoded but are class converted. Seems like an behavior to let the user handle the conversion when the type is declared in the plumber file.